reclist/metrics/novelty.py

```diff
@@ -0,0 +1,157 @@
+""" Collection of novelty-based metrics """
+from collections import defaultdict
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+def gini_index_at_k(y_preds, candidate_list, k: int=10, debug: bool=False) -> float:
+    no_items = len(set(candidate_list))
+    all_preds = []
+    for _p in y_preds:
+        all_preds += _p[:k]
+    all_preds_flattened = np.asarray(all_preds)
+    items, counts = np.unique(all_preds_flattened, return_counts=True)
+    counts.sort()
+    num_recommended_items = counts.shape[0]
+    total_num = len(y_preds)*k
+    idx = np.arange(no_items-num_recommended_items+1, no_items+1)
+    gini_index = np.sum((2*idx-no_items-1)*counts)/total_num
+    gini_index /= no_items
+    return gini_index
+
+
+def gini_index_at_k_user_differential(y_preds, y_test, candidate_list, k=10,
+                                      debug=False, user_feature='age_range', **kwargs):
+    breakdown = _breakdown_preds_by_user_feature(y_test, y_preds,
+                                                 user_feature=user_feature)
+    retval = _apply_func_to_breakdown(gini_index_at_k, breakdown, candidate_list,
+                                    k=k, debug=debug)
+    if debug:
+        plot_results_breakdown(retval, f"GiniIndex@{k}_by_{user_feature}.pdf", f"GiniIndex@{k}",
```

Comment:
@DWalker487 could you please change the destination path for those plots? 
It's used like this from RecList:
https://github.com/bbc/datalab-reclist/blob/3ace633246563abb9e6275f53bc93813e9b86c4a/reclist/metrics/hits.py#L43
They use `current.report_path` which is actually an auto-generated local folder `datalab-reclist/examples/.reclist/BBCSoundsRecList/BBCSoundsLightFMSimulatorModel` where you can find a `plots` folder from each run.